### PR TITLE
infraapi: correct condition type

### PR DESCRIPTION
**Pull request by zirain**
**What type of PR is this?**

/kind api-change
/kind bug

**What this PR does / why we need it**:

`capiv1.Conditions` alias to `[]capiv1.Condition`, use the right type.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: pkg/apis/infra/v1alpha1/cluster_types.go [hunk 1]
```diff
@@ -226,13 +226,27 @@ type ResourceRef struct {
...
 	Phase string `json:"phase,omitempty"`
 	// APIEndpoint is the endpoint to communicate with the apiserver.
+	// +optional
 	APIEndpoint string `json:"apiEndpoint,omitempty"`
 	// ServiceAccountIssuer is the URL of the service account issuer.
+	// +optional
 	ServiceAccountIssuer string `json:"serviceAccountIssuer"`
+	// ObservedGeneration is the most recent generation observed by the controller.
+	// +optional
+	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
```
Comment (hzxuzhonghu) [inline pkg/apis/infra/v1alpha1/cluster_types.go:241]: what does this mean?

Comment (zirain) [reply pkg/apis/infra/v1alpha1/cluster_types.go:241]: https://alenkacz.medium.com/kubernetes-operator-best-practices-implementing-observedgeneration-250728868792

Comment (hzxuzhonghu) [reply pkg/apis/infra/v1alpha1/cluster_types.go:241]: It is still ambiguous while looking at the link to Cluster API. Actually in that PR no controller is setting up `ObservedGeneration`

Comment (hzxuzhonghu) [reply pkg/apis/infra/v1alpha1/cluster_types.go:241]: And what could it show to users or controllers

**Review by hzxuzhonghu — APPROVED** [review]

**Pull request merged**